### DISCUSSION TITLE
Forward simulation uncertainty range

### DISCUSSION THREAD

**Discussion by cnicol-gwlogic** [pastas/pastas#398] [Q&A] [answered]
Hi there,

Firstly - thanks very much for a very nice piece of software. Much appreciated.

My query: I might be missing something dead obvious (sorry) , but is there a method for plotting simulation forecast uncertainties (based on the history match)? I can't see how to obtain this for periods beyond the observation record, other than using the emceehammer (ha! which I have done, but would like to keep things simpler if I can).

Many thanks.
Chris.

**Comment by dbrakenhoff** [comment, marked as answer]
Hi Chris,

I'm not entirely sure if this is what you're looking for but you should be able to pass `tmin` and `tmax` to the calculation of the prediction interval. This is the uncertainty as a result of estimated parameter uncertainties and the standard deviation of the residuals (for the calibration period). 

For example:

```python
ml  # some model
ml.solve(tmax="2015")
pi = ml.fit.prediction_interval(tmin="2015", tmax="2022")
```

Maybe that helps?

Davíd

**Reply by cnicol-gwlogic** [reply]
Brilliant @dbrakenhoff - thanks a lot, that works perfectly (apols, I should have noticed that).

**Comment by nelerey** [comment]
Maybe it'd be worth emphasizing in the documentation and/or uncertainty examle that tmin and tmax can be passed to `pastas.solver.BaseSolver.prediction_interval()` in the `**kwargs`? 
(suggesting this because I just found this closed discussion after forgetting how to do this and not finding the answer in the docs.)

**Reply by dbrakenhoff** [reply]
Done! See #785.
